packages/bundler-plugin-core/src/options-mapping.ts

```diff
@@ -81,7 +80,7 @@ export function normalizeUserOptions(userOptions: UserOptions): InternalOptions
     project: userOptions.project,
     authToken: userOptions.authToken,
     url: userOptions.url ?? "https://sentry.io/",
-    release: userOptions.release ?? detectRelease(),
+    release: userOptions.release ?? "",
```

Comment:
This seems odd but I want to keep `release` a required property, hence I'm assigning an empty string.

Reply:
Would it hurt us to make it an optional property? If it is actually optional now, we should define it as such. (Same probably down the line for project, authToken etc. because they could be defined via config/env-vars now right?)

Reply:
`release` is a special case here because we can't easily await the `cli.releases.proposeVersion` promise in the top-level function we pass to `createUnplugin`. Hence I moved it to the `beforeBuild` hook

Reply:
> because they could be defined via config/env-vars now right?

@lforst especially the config file is a valid point. Haven't thought about this when making the change here. So this means that, unless we want to read and parse the config file during normalization (which I don't want to do unless necessary), we have to refactor our internal options conversion to actually make these parameters (releaes, authtoken, etc) really optional... 

I'm gonna merge this PR to get some of the quick CLI replacements in and then tackle the options again. In the end this will change this line above